src/pages/BusPage/hooks/useBusLeftTime.ts

```diff
@@ -0,0 +1,43 @@
+import { getBusInfo } from 'api/bus';
+import { useQueries } from 'react-query';
+import { BUS_TYPES } from 'static/bus';
+
+const BUS_KEY = 'bus_info';
+
+interface Props {
+  departList: string[];
+  arrivalList: string[];
+}
+
+const emptyRouteData = {
+  now_bus: {
+    remain_time: '미운행',
+    bus_number: null,
+  },
+  next_bus: {
+    remain_time: '미운행',
+    bus_number: null,
+  },
+} as const;
+
+const useBusLeftTIme = ({ departList, arrivalList }: Props) => {
+  const queries = BUS_TYPES.map((type, idx) => ({
+    queryKey: [BUS_KEY, type, departList[idx], arrivalList[idx]],
+    queryFn: () => getBusInfo(type, departList[idx], arrivalList[idx]),
```

Comment:
해당 함수는 밑과 같이 바꿀 수 있겠네요.
```suggestion
    queryFn: ([_, busType, department, arrival]) => getBusInfo(busType, department, arrival),
```

Reply:
[쩝쩝박사 프로젝트에서 남긴 코멘트](https://github.com/BCSDLab/JJBAKSA_FRONT_END/pull/42#discussion_r1048010583) 내용과 더불어, queryKey의 타입이 `unknown[]`으로 정해져있어 `string` 타입에 맞출 수 없습니다.
타입 이슈를 해결할 방법이 있을까요?
```tsx
  const queries: UseQueryOptions[] = BUS_TYPES.map((type, idx) => ({
     // ~~~~~~~ 'queryFn' 타입이 올바르지 못함
    queryKey: [BUS_KEY, type, departList[idx], arrivalList[idx]],
    queryFn:
      (
        { queryKey: [busType, department, arrival] }: { queryKey: string[] },
      ) => getBusInfo(busType, department, arrival),
```

Reply:
해당 코멘트 참고해주세요 :)

Reply:
`unknown[]`으로 나오는 문제는 타입을 직접 지정해서 해결할 수 있었네요. 수정했습니다. [c055489](https://github.com/BCSDLab/KOIN_WEB_RECODE/pull/24/commits/c0554894e5e3275e777faecfdabf351cc32c75b5)
```tsx
const queries = BUS_TYPES.map<QueryOptions<BusResponse, AxiosError, BusResponse, string[]>>(
```